src/main/java/com/google/firebase/auth/Tenant.java

```diff
@@ -134,14 +161,20 @@ public static Builder newBuilder() {
      * Returns a builder for a tenant update request.
      */
     public static Builder newBuilder() {
+      // TOOD(micahstairs): Figure out if any of the UpdateRequest fields should default to anything
+      // particular. If any fields do not have a reasonable default, then the
+      // fields should be made nullable, otherwise AutoValue will throw an
+      // exception if those fields are left unset.
```

Comment:
I think this shouldn't be exposed. `Builder` seems to be a package-protected. Also we don't have a similar method in `UserRecord.UpdateRequest`. May be just expose the constructor.

```
Tenant tenant = ...;
UpdateRequest req = tenant.updateRequest();

// Or alternatively users should be able to:
UpdateRequest req = new Tenant.UpdateReuqst(tenant.getTenantId());
```

Reply:
Hmm.. I gave this some thought and ended up rewriting the create/update classes without using AutoValue. This is a lot more consistent with UserRecord's request classes. It will also make it easier to serialize these into json requests.